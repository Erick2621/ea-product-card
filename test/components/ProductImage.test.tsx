import { describe, expect, test } from "vitest";
import ProductCard, { ProductImage } from "../../src/components";
import { product1, product2 } from "../data/products";
import { render } from "@testing-library/react";
import { domToJSON } from "../functions/domToJSON";


describe('ProductImage', () => {
    test('debe de mostrar el componente correctamente con la imagen personalizada', () => {

        const { container } = render(<ProductImage img="https://hola.jpg" />);

        // Convertir el primer nodo hijo (el span) a JSON
        const json = domToJSON(container.firstChild as HTMLElement);
        console.log(json);

        expect(json).toMatchSnapshot();
    });

    test('debe de mostrar el componente  con la imagen del producto', () => {

        const { container } = render(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        );

        // Convertir el primer nodo hijo (el span) a JSON
        const json = domToJSON(container.firstChild as HTMLElement);
        console.log(JSON.stringify(json, null, 2));


        expect(json).toMatchSnapshot();
    });
});