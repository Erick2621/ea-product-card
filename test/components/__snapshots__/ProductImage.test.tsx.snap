// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductImage > debe de mostrar el componente  con la imagen del producto 1`] = `
{
  "props": {
    "children": [
      {
        "props": {
          "alt": "Product Image",
          "className": "_productImg_3e9733 undefined",
          "src": "./coffee-mug.png",
        },
        "type": "img",
      },
    ],
    "className": "_productCard_3e9733 undefined",
  },
  "type": "div",
}
`;

exports[`ProductImage > debe de mostrar el componente correctamente con la imagen personalizada 1`] = `
{
  "props": {
    "alt": "Product Image",
    "className": "_productImg_3e9733 undefined",
    "src": "https://hola.jpg",
  },
  "type": "img",
}
`;
